src/sentry/rules/actions/base.py

```diff
@@ -1,10 +1,37 @@
 from __future__ import absolute_import, print_function
 
+import functools
+import logging
+import operator
+import six
+
+from django import forms
+
 from sentry.constants import ObjectStatus
 from sentry.models.integration import Integration
 from sentry.models import ExternalIssue, GroupLink
 from sentry.rules.base import RuleBase
 
+logger = logging.getLogger("sentry.rules")
+
+INTEGRATION_KEY = "integration"
+
+
+class IntegrationNotifyServiceForm(forms.Form):
+    integration = forms.ChoiceField(choices=(), widget=forms.Select())
+
+    def __init__(self, *args, **kwargs):
+        integrations = [(i.id, i.name) for i in kwargs.pop("integrations")]
+        super(forms.Form, self).__init__(*args, **kwargs)
+        if integrations:
+            self.fields[INTEGRATION_KEY].initial = integrations[0][0]
+
+        self.fields[INTEGRATION_KEY].choices = integrations
+        self.fields[INTEGRATION_KEY].widget.choices = self.fields[INTEGRATION_KEY].choices
+
+    def clean(self):
```

Comment:
@ceorourke You don't need to define the `clean` function here if all you do is call `super`. It's not doing anything.